### PR TITLE
Add option to allow users to specify how they woudl like properties to be implemented.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/5898

We now provide an option to let a user say if they would prefer properties to be generated as 'auto properties' or as 'throwing properties'.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide .   Implement-abstract-type support coming shortly.

**Comment by Pilchie** [comment]
Along or instead of this, I'd love to see us add an "Introduce field and convert to full property".  Then I would have the default be to use auto-props everywhere, but if people need to add logic, it's easy to convert...

**Comment by CyrusNajmabadi** [comment]
Whatever we do, i want the option of "generate throwing bodies", because i definitely make use of that to ensure that i have to think about what actual logic needs to go in the property.   If it becomes an auto-prop, then it's too simple to just have it be stubbed out, and not realize i need to go actually implement the property properly.

**Comment by CyrusNajmabadi** [comment]
> Along or instead of this, I'd love to see us add an "Introduce field and convert to full property"

I agree.  To me that seems like a complimentary feature.  Just along the general lines of "i'm working with my code, and i've now realized i want my property to be more full featured."

**Comment by CyrusNajmabadi** [comment]
That also seems like a nice feature for @cbo425 to potentially do next!

**Comment by sharwell** [comment]
@Pilchie Is there not a Fix All in Document available for converting an auto-property to a full property?

**Comment by CyrusNajmabadi** [comment]
Need review plz.

File: src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceTests.vb [hunk 1]
```diff
@@ -4518,5 +4519,42 @@ Namespace System
...
+class Class
+    implements [|IInterface|]
+end class",
+"interface IInterface
+    readonly property ReadOnlyProp as integer
+    property ReadWriteProp as integer
+    writeonly property WriteOnlyProp as integer
+end interface
+
+
```
Comment (Pilchie) [inline src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceTests.vb]: Why is there an extra blank line here compared to the input?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceTests.vb]: no reason.  :)

**Comment by CyrusNajmabadi** [comment]
@ManishJayaswal 

**Customer scenario**

This is an improvement to our code generation experience whereby we give users a new option to control how they want properties to be generated.  i.e. using more modern C# property syntax vs the older existing style.

**Bugs this fixes:**

Fixes #5898

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

No root cause.  This is feature work.

**How was the bug found?**

Customer and internally requested.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz This is feature-work, but one that we have customer requests for and which fits with making teh IDE work better for more modern C# constructs.

**Pull request merged**